lib/aur-depends

```diff
@@ -197,7 +197,7 @@ if cd "$tmp" && mkdir json tsv; then
             ;;
         # pkgbase (AUR, graph)
         pkgbase_graph)
-            package_graph 'pkgbase' "$deptable" "$deptable" | sort -k1b,1
+            package_graph 'pkgbase' "$deptable" "$deptable" | sort -k1b,1 | uniq
```

Comment:
`sort -k1,1b -k1 -u`, no?

Reply:
Is there a difference between `sort -k1,1b` and `sort -k1b,1`?

Reply:
The key definiton is `field[type],field[type]`, so if you had used `-k1b,2n` it would sort field 1 while ignoring leading blanks and field 2 numerically.  As your range is `1,1` it doesn't matter which field gets `b` as they're both the same.  That is my reading the standard.  

<details>
    <summary>In fact it gets more complicated, from sort(1p)</summary>
<pre>
                 When  there are multiple key fields, later keys shall be com‐
                 pared only after all earlier keys compare equal. Except  when
                 the  -u  option  is  specified,  lines that otherwise compare
                 equal shall be ordered as if none of the options -d, -f,  -i,
                 -n,  or  -k  were present (but with -r still in effect, if it
                 was specified) and with all bytes in the lines significant to
                 the  comparison.  The order in which lines that still compare
                 equal are written is unspecified.

...

EXTENDED DESCRIPTION
       The notation:

           -k field_start[type][,field_end[type]]

       shall  define  a  key  field  that  begins  at  field_start and ends at
       field_end inclusive, unless field_start falls beyond  the  end  of  the
       line  or after field_end, in which case the key field is empty. A miss‐
       ing field_end shall mean the last character of the line.

       A field comprises a maximal sequence of non-separating characters  and,
       in the absence of option -t, any preceding field separator.

       The  field_start  portion  of the keydef option-argument shall have the
       form:

           field_number[.first_character]

       Fields and characters within fields shall be numbered starting with  1.
       The  field_number  and  first_character pieces, interpreted as positive
       decimal integers, shall specify the first character to be used as  part
       of  a  sort  key. If .first_character is omitted, it shall refer to the
       first character of the field.

       The field_end portion of the  keydef  option-argument  shall  have  the
       form:

           field_number[.last_character]

       The  field_number  shall  be  as  described above for field_start.  The
       last_character piece, interpreted as a  non-negative  decimal  integer,
       shall specify the last character to be used as part of the sort key. If
       last_character evaluates to zero  or  .last_character  is  omitted,  it
       shall  refer to the last character of the field specified by field_num‐
       ber.

       If the -b option or b type modifier is in effect, characters  within  a
       field  shall  be counted from the first non-<blank> in the field. (This
       shall apply separately to first_character and last_character.)

...

        *  The  output  of  sort (without -u) can contain identical lines that
           are not adjacent, if it does not implement the recommended  further
           byte-by-byte comparison of lines that collate equally. This affects
           the use of sort with comm and uniq; see the APPLICATION  USAGE  for
           those utilities.
</pre>
</details>

Reply:
That's definitely more complex than I anticipated. For your examples,
```
sort -b -k1,1 -k1 -u
sort -k1b,1 -k1 -u
sort -k1,1b -k1 -u
```
all seem to work though (`-r` probably doesn't matter here, since the edges of the graph are the same)

Reply:
I only added `-r` to see how it might interact with subsequent `-k` and `-u` flags.